### ISSUE TITLE
[java] Array<Int>.toString() prints null instead of 0 for uninitialized elements.

### ISSUE THREAD

**Issue by sebthom** [HaxeFoundation/haxe#6536]
In the example below I would expect to get [ 1, 0, 2 ] like with C++, C# and HL.
```haxe
var a:Array<Int> = [1];
a[2] = 2;
trace(a);    // prints: [ 1, null, 2 ]
trace(a[1]); // prints: 0
```

**Comment by Simn** [comment]
@waneck: Any idea why this behaves differently between Java and C#?

**Comment by nanjizal** [comment]
I don't know about c# but it's effectively internally it's   NativeArray < HxObject > and not really NativeArray < Int > so the way it defaults is different from how might be expected:

 "Arrays of primitives have elements that are initialized to default values. Arrays of objects have the value null in each element." http://www.pearsonitcertification.com/articles/article.aspx?p=101368&seqNum=7

```
// var a: Array<Int> = [1];
haxe.root.Array<java.lang.Object> a = new haxe.root.Array<java.lang.Object>(new java.lang.Object[]{1});
```

**Comment by Simn** [comment]
The Java target will be put in "regression fixes only" state with the Haxe 4.3.0 release. See https://github.com/HaxeFoundation/haxe/wiki/Target-Support-Level#java-and-c-since-430 for additional information. We apologize for any inconvenience this causes!
